language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConfigurableCodeAction.java

```diff
@@ -0,0 +1,346 @@
+        NodeList<ImportDeclarationNode> importsList = modulePartNode.imports();
+
+        ModuleVariableDeclarationNode configurableVarNode = null;
+        for (Node node: modulePartNode.children()) {
+            if (node.kind() == SyntaxKind.MODULE_VAR_DECL) {
+                ModuleVariableDeclarationNode modVarDeclarationNode = (ModuleVariableDeclarationNode) node;
+                if (hasConfigurableQualifier(modVarDeclarationNode)) {
+                    configurableVarNode = modVarDeclarationNode;
+                    break;
+                }
+            }
+        }
+
+        if (configurableVarNode != null) {
+            return new ConfigurableData(new Position(configurableVarNode.lineRange().startLine().line(), 0),
+                    !configurableVarNode.toString().startsWith(System.lineSeparator()));
+        } else if (importsList.isEmpty()) {
+            return new ConfigurableData(
+                    PositionUtil.toPosition(modulePartNode.lineRange().startLine()), false);
+        }
+        ImportDeclarationNode lastImport = importsList.get(importsList.size() - 1);
+        return new ConfigurableData(
+                new Position(lastImport.lineRange().endLine().line() + 1, 0), true);
+    }
+
+    private static String getConfigurableName(CodeActionContext context) {
+        Position pos = context.range().getEnd();
+        Set<String> allNames = context.visibleSymbols(new Position(pos.getLine(), pos.getCharacter())).stream()
+                .map(Symbol::getName)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toSet());
+
+        return NameUtil.generateTypeName(CONFIGURABLE_NAME_PREFIX, allNames);
+    }
+
+    private static String getConfigurableNameUsingParamName(CodeActionContext context, String paramName) {
+        Position pos = context.range().getEnd();
+        Set<String> allNames = context.visibleSymbols(new Position(pos.getLine(), pos.getCharacter())).stream()
+                .map(Symbol::getName)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toSet());
+
+        return NameUtil.generateTypeName(paramName, allNames);
+    }
+
+    private static boolean hasConfigurableQualifier(ModuleVariableDeclarationNode modVarDeclarationNode) {
+        return modVarDeclarationNode.qualifiers()
+                .stream().anyMatch(q -> q.text().equals(Qualifier.CONFIGURABLE.getValue()));
+    }
+
+    private static boolean hasMultipleArgsSelected(Node node) {
+        return argumentSyntaxList.contains(node.kind());
+    }
+
+    private static boolean withInRange(Node argument, Range range) {
+        return PositionUtil.isWithinRange(PositionUtil.toPosition(argument.lineRange().startLine()), range) ||
+                PositionUtil.isWithinRange(PositionUtil.toPosition(argument.lineRange().endLine()), range);
+    }
```

Comment:
Can't we use  `PositionUtil`'s `isWithinLineRange(LineRange lineRange, LineRange enclosingRange)` instead this

Reply:
I guess then we have to convert the range to a line range.